src/main/msp/msp_serial.c

```diff
@@ -527,7 +549,9 @@ void mspSerialProcess(mspEvaluateNonMspData_e evaluateNonMspData, mspProcessComm
                 const bool consumed = mspSerialProcessReceivedData(mspPort, c);
 
                 if (!consumed && evaluateNonMspData == MSP_EVALUATE_NON_MSP_DATA) {
-                    mspEvaluateNonMspData(mspPort, c);
+                    if (mspEvaluateNonMspData(mspPort, c)) {
+                        break;
```

Comment:
Ugly way to handle this is `portIndex=--; break;`

A bit better, but still far from elegant:
```
for (uint8_t portIndex = 0, retryPort = false; portIndex < MAX_MSP_PORT_COUNT; portIndex += retryPort ? 0 : 1, retryPort = false) {

...
    retryPort = true;
    break;
```

Or simply :
```
`goto retryPort;`

Reply:
Yes, much better than these hacks